OpenSim/Analyses/InverseDynamics.cpp

```diff
@@ -344,10 +344,11 @@ record(const SimTK::State& s)
         const Coordinate& coord = *coordinates[_accelerationIndices[i]];
         int ind = _statesStore->getStateIndex(coord.getSpeedName(), 0);
         if (ind < 0){
-            string fullname = coord.getJoint().getName() + "/" + coord.getSpeedName();
+            // get the full coordinate speed state variable path name
+            string fullname = coord.getStateVariableNames()[1];
```

Comment:
This is more reliable than before. The state variable names are always listed with the Component's added states first. Only changing the order of coordinate *value* and *speed* would break this.

Reply:
Okay thanks for explaining.

Reply:
This still might not be efficient if there are subcomponents, because the array of state variable names is not cached.

Reply:
> This still might not be efficient if there are subcomponents, because the array of state variable names is not cached.

Possibly, but this Analysis is no longer used to perform InverseDynamics. It is only here for the mass adjustment step of RRA. The whole name lookup every step is kind of absurd to begin with, but not worth fixing a virtually unused class.